app/gui/src/controller/searcher.rs

```diff
@@ -477,17 +478,20 @@ impl Data {
 /// existing node).
 #[derive(Clone, CloneRef, Debug)]
 pub struct Searcher {
-    logger:           Logger,
-    data:             Rc<RefCell<Data>>,
-    notifier:         notification::Publisher<Notification>,
-    graph:            controller::ExecutedGraph,
-    mode:             Immutable<Mode>,
-    database:         Rc<model::SuggestionDatabase>,
-    language_server:  Rc<language_server::Connection>,
-    ide:              controller::Ide,
-    this_arg:         Rc<Option<ThisNode>>,
+    logger: Logger,
+    data: Rc<RefCell<Data>>,
+    notifier: notification::Publisher<Notification>,
+    graph: controller::ExecutedGraph,
+    mode: Immutable<Mode>,
+    database: Rc<model::SuggestionDatabase>,
+    language_server: Rc<language_server::Connection>,
+    ide: controller::Ide,
+    this_arg: Rc<Option<ThisNode>>,
     position_in_code: Immutable<Location>,
-    project:          model::Project,
+    project: model::Project,
+    /// A component list builder with favorites already set to
+    /// [`controller::ExecutedGraph::component_groups`].
```

Comment:
I'm not sure I'm following this comment. The favorites are already set to something here, but where in the code they are not set yet? If we refer here as "already", it means that there is some code where they are not yet. Maybe we should link this code? Without this information, I feel this comment is incomplete.

Reply:
I think the "already" meaning here is "you don't need to do it anymore" rather than "there is code where they're not set". Perhaps the term "prepopulated" will suit here.

I would also add an info _why_ we keep such builder.